core/system_impl.cpp

```diff
@@ -631,95 +631,131 @@ void SystemImpl::set_param_ext_int_async(const std::string &name, int32_t value,
     _params.set_param_async(name, param_value, callback, true);
 }
 
-std::pair<bool, float> SystemImpl::get_param_float(const std::string &name)
+std::pair<MAVLinkParameters::Result, float> SystemImpl::get_param_float(const std::string &name)
 {
-    auto prom = std::make_shared<std::promise<std::pair<bool, float>>>();
+    auto prom = std::make_shared<std::promise<std::pair<MAVLinkParameters::Result, float>>>();
```

Comment:
I don't think it has to be a pointer, as it is used only within the scope of this function. Unless the callback can live longer than the function?

Reply:
Hm, you're right. I guess that's just what the very first example of promise/future looked like. I'll fix it.

Reply:
Yeah, we have many places that use pointers where that wouldn't be necessary (typically the examples or integration tests). Not a very big deal, but we can think about changing that when we see them :-).